src/plugins/kdb/db2/libdb2/include/db.hin

```diff
@@ -40,6 +40,10 @@
 
 #include <sys/types.h>
 
+#ifdef __HAIKU__
+#include <stdint.h>
+#endif
```

Comment:
No, it's the definition of `u_int32_t`:
```
~> grep -rn /system/develop/headers/posix -e u_int32_t
stdint.h:163:typedef uint32_t u_int32_t;
```
Caused by
```
In file included from ./../include/db-int.h:40,
                 from hash_bigkey.c:67:
../../../../../include/db.h:161:2: error: expected specifier-qualifier-list before 'u_int32_t'
  u_int32_t  /* hash function */
```
Where `db-int.h` is generated from the `db.hin` file in the commit.

I've been taking the approach of not breaking other platforms, hence the defines. An alternative may be the larger <inttypes.h> header?

Reply:
That is, `<inttypes.h>` includes `<stdint.h>` plus extra definitions.

Reply:
u_int32_t is supposed to be defined if necessary by db-config.h which has:

    #ifndef HAVE_U_INT32_T
    #define u_int32_t unsigned int
    #endif

Where HAVE_U_INT32_T is in turn defined or not defined according to AC_CHECK_TYPES().  So I guess the problem is that AC_CHECK_TYPES() includes stdint.h (among other things; see https://www.gnu.org/software/autoconf/manual/autoconf-2.63/html_node/Default-Includes.html#Default-Includes ) but this ancient header does not.  Haiku might have a C conformance problem here since stdint.h is a standard header and u_int32_t isn't a standard type name, but that's probably not a fruitful line of investigation.

I'll accept including stdint.h here unconditionally, since we require that level of C support for the build (it's included unconditionally in krb5.hin).  A more robust fix would involve using standard type names in the DB2 code, but I don't want to churn that code to such a degree.

Reply:
It's quite possible, our headers are a bit of a mess :-/

Anyway, pushed the requested change. Shall I rebase to remove the now redundant commits, or are you happy with a squash merge?